### PR TITLE
Migrate to package_info_plus

### PR DESCRIPTION
**Pull request by kuhnroyal**
I replaced the discontinued package and update the version numbers.
Should be ready to be released as is.

### CODE REVIEW

File: update_available/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.1.0
+
+- Migrate to `package_info_plus` for iOS
```
Comment (mateusfccp) [inline update_available/CHANGELOG.md:3]:
```suggestion
- Replace `package_info`, which is deprecated, with `package_info_plus`
```

File: update_available_ios/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 2.1.0
+
+- Migrate to `package_info_plus`
```
Comment (mateusfccp) [inline update_available_ios/CHANGELOG.md:3]:
```suggestion
- Replace `package_info`, which is deprecated, with `package_info_plus`
```

File: update_available_ios/pubspec.yaml [hunk 3]
```diff
@@ -12,11 +12,15 @@ dependencies:
     sdk: flutter
   http: ^0.13.0
   meta: ^1.3.0
-  package_info: ^2.0.0
-  update_available_platform_interface: ^2.0.0
+  package_info_plus: ^1.2.0
```
Comment (mateusfccp) [inline update_available_ios/pubspec.yaml:15]: The last version of `package_info_plus` is `1.4.2`. Is there a reason why you decided to use `^1.2.0` here?

File: update_available_android/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 2.1.0
+
+- No documented changes
+
```
Comment (mateusfccp) [inline update_available_android/CHANGELOG.md:4]: You can simply leave this out.

```suggestion
```

File: update_available_ios/pubspec.yaml [hunk 5]
```diff
@@ -12,11 +12,15 @@ dependencies:
...
   pub_semver: ^2.0.0
 
 dev_dependencies:
   flutter_test:
     sdk: flutter
   pedantic: ^1.11.0
+
+dependency_overrides:
+  update_available_platform_interface:
+    path: ../update_available_platform_interface
```
Comment (mateusfccp) [inline update_available_ios/pubspec.yaml:26]: Why is this override necessary?

File: update_available_platform_interface/CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,7 @@
+## 2.1.0
+
+- No documented changes
+
```
Comment (mateusfccp) [inline update_available_platform_interface/CHANGELOG.md:4]: You can simply leave this out.

```suggestion
```

**Review by mateusfccp — CHANGES_REQUESTED** [review]
Thanks for the PR!

Here are some considerations.

File: update_available_ios/pubspec.yaml (hunk 3 above)
Comment (kuhnroyal) [reply update_available_ios/pubspec.yaml:15]: Not really but all the other dependency bounds are pretty low/old as well.

File: update_available_android/CHANGELOG.md (hunk 4 above)
Comment (kuhnroyal) [reply update_available_android/CHANGELOG.md:4]: Ok, but no changelog requires a `--force` publish

File: update_available_ios/pubspec.yaml (hunk 3 above)
Comment (mateusfccp) [reply update_available_ios/pubspec.yaml:15]: Yeah, I have to take a time to update the others as well, but if there's no breakage we may use the latest version of `package_info_plus`.

Comment (kuhnroyal) [reply update_available_ios/pubspec.yaml:15]: I wouldn't increase it for now. If you plan a general dependency update then you can raise it there.
For now it is an easier upgrade path if we do not force the latest version.

**Review by mateusfccp — APPROVED** [review]
LGTM, thanks!

**Comment by kuhnroyal** [comment]
Thanks!

**Pull request merged**